clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7449,6 +7449,9 @@ let CategoryName = "Lambda Issue" in {
     "%select{| explicitly}1 captured here">;
   def err_implicit_this_capture : Error<
     "implicit capture of 'this' is not allowed for kernel functions">;
+  def warn_constexpr_kernel_arg : Warning<
+    "constexpr '%0' is captured and is not a compile-time constant">,
```

Comment:
How about something like this:
```suggestion
    "captured constexpr '%0' will appear as regular kernel argument in device code">,
```
It sounds a bit more clear

Reply:
I think we can omit 'regular' and just say something like - 

 "captured constexpr '%0' will be a kernel argument in device code"

I'm not too particular about this and so feel free to choose what you prefer.

Reply:
I'm fine with both variants.